### PR TITLE
WIFI-1864-CPU-Temperature_Update

### PR DESCRIPTION
**Pull request by NagConnectus**
This patch will add the support to read the cpu temperature of ath11k
chipset boards and update to the UI
Signed-off-by: Nagendrababu <nagendrababu.bonkuri@connectus.ai>

### CODE REVIEW

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/stats.c [hunk 1]
```diff
@@ -372,7 +372,10 @@ bool target_stats_device_temp_get(radio_entry_t *radio_cfg, dpp_device_temp_t *t
 
 	fclose(fp);
 	temp_entry->type  = radio_cfg->type;
-	temp_entry->value = temperature / 1000;
+	if(temperature > 125)
+		temp_entry->value = temperature / 1000;
+	else
+		temp_entry->value = temperature;
```
Comment (RickSommerville) [inline feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/stats.c:378]: This seems very arbitrary.  If there is a scaling difference between the two methods of obtaining temperature data, then make the scaling scaling change at the point of retrieval.  This way you will know explicitly which scale applies.

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/utils.c [hunk 2]
```diff
@@ -197,22 +197,32 @@ int phy_from_path(char *_path, char *phy, unsigned int idx)
...
+	snprintf(tmp_wifi6, sizeof(tmp_wifi6), "/sys/class/ieee80211/%s/cooling_device/subsystem/thermal_zone0/", phy);
+	if (!glob(tmp_wifi5, GLOB_NOSORT | GLOB_MARK, NULL, &gl)) {
+		if (gl.gl_pathc) {
+			strcpy(hwmon, gl.gl_pathv[0]);
+			strncat(hwmon, "temp1_input", PATH_MAX);
+			globfree(&gl);
+			return 0;
+		}
+	}
+	else if(glob(tmp_wifi6, GLOB_NOSORT | GLOB_MARK, NULL, &gl))
```
Comment (RickSommerville) [inline feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/utils.c:215]: Using glob() here and !glob() above at line 207 is bad for readability.

Really you have one operation made on two data sets:   
"/sys/class/ieee80211/%s/device/hwmon/*" with "temp1_input", and"/sys/class/ieee80211/%s/cooling_device/subsystem/thermal_zone0/" with "temp".

You could create a helper function that takes these two parameters as inputs, and &temp for the result, and that returns success/fail.  Then the code would be very easy to understand and you'd avoid the duplication and you'd have a helper function that could be used for other information (say CPU utilization maybe).

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/stats.c (hunk 1 above)
Comment (NagConnectus) [reply feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/stats.c:376]: done

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/utils.c (hunk 2 above)
Comment (NagConnectus) [reply feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/utils.c]: sure, done

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/utils.c [hunk 3]
```diff
@@ -195,22 +195,34 @@ int phy_from_path(char *_path, char *phy, unsigned int idx)
...
+    return 0;
+}
+
+int phy_find_hwmon(char *phy, char *hwmon, bool *DegreesNotMilliDegrees)
+{
+    char tmp[PATH_MAX];
+    *hwmon = '\0';
+    snprintf(tmp, sizeof(tmp), "/sys/class/ieee80211/%s/device/hwmon/*", phy);
+    if (!phy_find_hwmon_helper(tmp, "temp1_input", hwmon)) {
+    	*DegreesNotMilliDegrees=false;
```
Comment (RickSommerville) [inline feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/utils.c:217]: indentation

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/utils.c [hunk 4]
```diff
@@ -195,22 +195,34 @@ int phy_from_path(char *_path, char *phy, unsigned int idx)
...
+        return 0;
+    }
+    snprintf(tmp, sizeof(tmp), "/sys/class/ieee80211/%s/cooling_device/subsystem/thermal_zone0/", phy);
+    if (!phy_find_hwmon_helper(tmp, "temp", hwmon)) {
+    	*DegreesNotMilliDegrees=true;
```
Comment (RickSommerville) [inline feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/utils.c:222]: indentation

Comment (NagConnectus) [reply feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/utils.c]: done, thank you

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/utils.c (hunk 3 above)
Comment (NagConnectus) [reply feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/utils.c]: done

**Review by RickSommerville — APPROVED** [review]

**Comment by RickSommerville** [comment]
Merged 34c90062, thanks

**Pull request closed without merging**